csrc/runtime/fusion_kernel_runtime.cpp

```diff
@@ -756,6 +787,9 @@ void FusionKernelRuntime::compileKernel(
     SegmentedGroup* sg,
     hir::HostIrContainer* hic) {
   FUSER_PERF_SCOPE("FusionKernelRuntime::compileKernel");
+  c10::cuda::CUDAGuard dg(args.getDeviceIndex());
+  c10::Device device(c10::DeviceType::CUDA, args.getDeviceIndex());
```

Comment:
These two lines are moved from

https://github.com/NVIDIA/Fuser/pull/4373/files#diff-67ea17ca8b2cc538ed907fe9c9546fe45fe300237d1302379a14a0c0c0e79fcbL389

and 

https://github.com/NVIDIA/Fuser/pull/4373/files#diff-67ea17ca8b2cc538ed907fe9c9546fe45fe300237d1302379a14a0c0c0e79fcbL405-L406

I don't see any reason that they need to be done outside of `compileKernel`.